src/dynamodb.js

```diff
@@ -44,6 +31,11 @@ async function createTables (dynamoClient, uniqueIdentifier) {
   );
 
   if (failedProvisons.length) {
+    try {
+      await destroyTables(dynamoClient, uniqueIdentifier);
```

Comment:
why was it not necessary to destroy tables? Why didn't we need it before?

Reply:
The error I get when running LifeService unit tests is that createTables times out, or gets a connection reset after creating x tables, where x is not zero.  Then throws an exception when it hits here.  When the exception is thrown, the connection details aren't added to the test context so the afterEach throws an exception that it can't get the config details from context.dynamodb because it is null.

Then the next test tries to run, and runs into the error that x tables already exist and it can't delete them, so on and so forth for the rest of the tests using that docker image.

Reply:
yeah, I've definitely had some weirdness when running unit tests in lifie-service too. Seemed liked one test failing affected the rest of the tests

Reply:
It seems to me that `destroyTables` won't help in many cases. The ordering of both the create and the destroy is not well-determined. This means that if the create fails part way through the delete may also fail part way through and still leave state that will break subsequent tests.

Reply:
`destroyTables` does a lookup to see what tables actually exist, and matches those against the list of tables that it expects to delete.  If there were 2 tables it is tracking, one gets created and the second does not, then the delete code only deletes the one that was created.  I can add another test case demonstrating the issue.

Reply:
Ah, right. I missed that.